File: Python/level 5/How to line up.py
import math
def setAlign(n, k):
    list = [i for i in range(1,n+1)]
    answer = []
    for j in range(1,n+1):
        fac = math.factorial(n-1)
        print(k)
        answer.append(list.pop((k-1) // fac))
        n, k = n-1, k%fac
    return answer


# 아래는 테스트로 출력해 보기 위한 코드입니다.
print(setAlign(5, 24))

File: CPP/level 5/How to line up.cpp
#include<iostream>
#include<vector>
#include<cstring>
using namespace std;

int factorial(int n){
    int a = 1;
    for(int i=1; i<=n; i++)
        a *= i;
    return a;
}
vector<int> setAlign(int n, long long cnt)
{
	vector<int> answer;
    char arr[n];
    int fac=0,len = n;
    int idx = 1;
    for(int i=1; i<n+1; i++)
        arr[i-1] = i+48;
 
    for(int i=1; i<len+1; i++){
        fac = factorial(n-1);
        idx = (cnt-1) / fac;
        if(idx == -1)
            idx = n-1; 
        answer.push_back(arr[idx]-48);       
        memmove(arr+idx,arr+idx+1,strlen(arr)-idx);
        n -=1;
        cnt %= fac; 
    }
	return answer;
}
int main()
{
	int testn = 3;
	long long testcnt = 5;
	vector<int> testAnswer = setAlign(testn,testcnt);
	// 아래는 테스트로 출력해 보기 위한 코드입니다.

	for(int i=0; i< testAnswer.size(); i++)
	{
		cout << testAnswer[i] << " ";
	}
}
